File: app/src/main/java/com/uolinc/marvelapp/ui/characterlist/MainContrato.java
package com.uolinc.marvelapp.ui.characterlist;

import androidx.recyclerview.widget.LinearLayoutManager;

import com.uolinc.marvelapp.model.Result;

import java.util.List;

public interface MainContrato {

    interface View{
        void initialize();
        void setRecyclerViewCharacter();
        void setSpinnerOrderBy();
        void loadCharacterList(List<Result> _characterList);
        void showCharacterDetailActivity(Result result, String urlImage);
        void showError(boolean isLoading, int limit, int offset);
    }

    interface Presenter{
        String setOrderby(int position);
        void getData(int limit, int offset, String orderBy);
        void scrolledRecyclerViewCharacter(LinearLayoutManager linearLayoutManager, String orderBy);
    }
}

File: app/src/main/java/com/uolinc/marvelapp/ui/characterlist/MainContrato.kt
package com.uolinc.marvelapp.ui.characterlist

import com.uolinc.marvelapp.model.Result

interface MainContrato {

    interface View {
        fun initialize()
        fun setRecyclerViewCharacter()
        fun setSpinnerOrderBy()
        fun loadCharacterList(_characterList: List<Result>)
        fun showCharacterDetailActivity(result: Result, urlImage: String)
        fun showError(isLoading: Boolean, limit: Int, offset: Int)
    }

    interface Presenter {
        fun setOrderby(position: Int): String
        fun setParametersToGetData(visibleItemCount: Int, totalItemCount: Int,
                                   firstVisibleItemPosition: Int, orderBy: String)

        fun getData(limit: Int, offset: Int, orderBy: String)
    }
}
